_component_design/text-inputs.md

```diff
@@ -13,185 +13,291 @@ last-modified: 2017-08-17
+
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+### Basic layout
+ * Labels display the type of input a field requires. Every text field should have a label.
+ * Labels are placed above and are left aligned with the input box.
+ * Labels should always be visible.  Avoid using placeholder text for labelling.
+
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/text-fields/text-fields-text-input-basic-layout.svg)
+{% endfigure %}
+
+{% endcolumn %}
+</div>
+
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+### Placeholder text
+ * Placeholder text can be displayed in an empty text field until input is entered.
+ * The placeholder may be instructive text or an example, such as a phone number or email address.
+
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/text-fields/text-fields-text-input-placeholder-text.svg)
+{% endfigure %}
+
+{% endcolumn %}
+</div>
+
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+### Helper text
+ * Helper text should provide clear instructions for populating the input, such as examples, syntax, or other parameters 
+ * Keep text to a single line if possible, though two lines are acceptable
+ * If more than two lines of help text are needed, consider using a tooltip instead
+
+
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/text-fields/text-fields-text-input-helper-text.svg)
+{% endfigure %}
+
+{% endcolumn %}
+</div>
+
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+### Required fields
+ * To indicate that a field is required, display an asterisk (*) to the left of the label
+ * To aid recognition, the asterisk should overhang the space to the left of the group by 4 pixels
+
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/text-fields/text-fields-text-input-required.svg)
+{% endfigure %}
+
+{% endcolumn %}
+</div>
+
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+### Character or word counter
+ * Use character or word counters if there is a character or word limit on the input.
+ * The counter is right justified within the control grop
+ * Displayed as a ratio of characters used and the character limit
+ * format: characters used / character limit
```

Comment:
Editorial: Parallel structure of list elements:
```markdown
Format as `characters used/character limit`. For example, `5/10`.
```

Reply:
👍